packages/sit-onyx/src/components/OnyxStepper/OnyxStepper.vue

```diff
@@ -57,19 +53,54 @@ const value = computed({
   set: (value) => emit("update:modelValue", value),
 });
 
-const handleClick = (direction: "stepUp" | "stepDown") => {
-  if (!inputRef.value) return;
+function countDecimalPlaces(num: number) {
+  const numStr = num.toString();
+  if (numStr.includes(".")) {
+    return numStr.split(".")[1].length;
+  }
+  return 0;
+}
 
-  inputRef.value[`${direction}`]();
+const displayValue = computed(() => {
+  const precision = countDecimalPlaces(props.precision);
+  return value.value?.toFixed(precision);
+});
 
-  const newValue = inputRef.value.valueAsNumber;
-  value.value = isNaN(newValue) ? undefined : newValue;
+const handleClick = (direction: "stepUp" | "stepDown") => {
+  if (!inputRef.value) return;
+  const currentValue = inputRef.value.valueAsNumber || 0;
+  const step = props.stepSize ? props.stepSize : props.precision;
+  const stepValue = direction === "stepUp" ? step : -step;
+  const newValue = currentValue + stepValue;
+  const roundedValue = Math.round(newValue / props.precision) * props.precision;
+  if (props.min !== undefined && roundedValue < props.min) {
+    value.value = props.min;
+  } else if (props.max !== undefined && roundedValue > props.max) {
+    value.value = props.max;
+  } else {
+    value.value = roundedValue;
+  }
+};
+const handleChange = () => {
+  if (!inputRef.value) return;
+  const newValue = parseFloat(inputRef.value.value);
+  const precision = countDecimalPlaces(props.precision);
+  if (props.stripStep && newValue % props.precision !== 0) {
+    inputRef.value.value = value.value !== undefined ? value.value.toString() : "";
+  } else {
+    value.value = isNaN(newValue) ? undefined : parseFloat(newValue.toFixed(precision));
+    inputRef.value.value = newValue.toFixed(precision);
+    wasTouched.value = true;
+  }
 };
 
-const incrementLabel = computed(() => t.value("stepper.increment", { stepSize: props.step }));
-const decrementLabel = computed(() => t.value("stepper.decrement", { stepSize: props.step }));
+const incrementLabel = computed(() =>
+  t.value("stepper.increment", { stepSize: props?.stepSize ? props.stepSize : props.precision }),
```

Comment:
Please use a `computed` to determine the stepSize, so the logic and default value have a single source of truth:
```ts
const determinedStepSize = computed(() =>  props.stepSize ?? props.precision);
```